x-pack/plugins/apm/public/components/app/service_details/service_icons/index.tsx

```diff
@@ -78,12 +90,9 @@ export function ServiceIcons({ serviceName }: Props) {
         });
       }
     },
-    [isPopoverOpen, serviceName, start, end, uiFilters]
+    [selectedIconPopover, serviceName, start, end, uiFilters]
```

Comment:
Is the request made every time the popover is opened/closed or only the first time?

Reply:
The request is going to be cached, so no, the details are going to be fetched once and used until the parameters are changed. Take a look:

![ezgif com-gif-maker (3)](https://user-images.githubusercontent.com/55978943/102365313-b26d3a80-3fb7-11eb-8d21-14f9e5672e61.gif)